packages/components/src/mobile/inserter-button/index.native.js

```diff
@@ -6,115 +6,107 @@ import { View, TouchableHighlight, Text } from 'react-native';
+		  __( '%s block' );
+	const accessibilityLabel = sprintf( accessibilityLabelFormat, item.title );
 
-		const clipboardBlockStyles = getStylesFromColorScheme(
-			styles.clipboardBlock,
-			styles.clipboardBlockDark
-		);
-
-		const isClipboardBlock = item.id === 'clipboard';
-		const blockTitle = isClipboardBlock ? __( 'Copied block' ) : item.title;
-		const blockIsNew = item.isNew === true;
-		const accessibilityLabelFormat = blockIsNew
-			? // translators: Newly available block name. %s: The localized block name
-			  __( '%s block, newly available' )
-			: // translators: Block name. %s: The localized block name
-			  __( '%s block' );
-		const accessibilityLabel = sprintf(
-			accessibilityLabelFormat,
-			item.title
-		);
-
-		return (
-			<TouchableHighlight
-				style={ [
-					styles.touchableArea,
-					item.isDisabled ? styles.disabled : null,
-				] }
-				underlayColor="transparent"
-				activeOpacity={ 0.5 }
-				accessibilityRole="button"
-				accessibilityLabel={ accessibilityLabel }
-				onPress={ this.onPress }
-				disabled={ item.isDisabled }
-			>
-				<View style={ [ styles.modalItem, { width: maxWidth } ] }>
-					<View
-						style={ [
-							modalIconWrapperStyle,
-							itemWidth && {
-								width: itemWidth,
-							},
-							isClipboardBlock && clipboardBlockStyles,
-						] }
-					>
-						{ blockIsNew && (
-							<Icon
-								icon={ sparkles }
-								style={ styles.newIndicator }
-							/>
-						) }
-						<View style={ modalIconStyle }>
-							<Icon
-								icon={ item.icon.src || item.icon }
-								fill={ modalIconStyle.fill }
-								size={ modalIconStyle.width }
-							/>
-						</View>
+	return (
+		<TouchableHighlight
+			style={ [
+				styles.touchableArea,
+				item.isDisabled ? styles.disabled : null,
+			] }
+			underlayColor="transparent"
+			activeOpacity={ 0.5 }
+			accessibilityRole="button"
+			accessibilityLabel={ accessibilityLabel }
+			onPress={ onPress }
+			disabled={ item.isDisabled && ! item.disabledMessage }
+		>
+			<View style={ [ styles.modalItem, { width: maxWidth } ] }>
+				<View
+					style={ [
+						modalIconWrapperStyle,
+						itemWidth && {
+							width: itemWidth,
+						},
+						isClipboardBlock && clipboardBlockStyles,
+					] }
+				>
+					{ blockIsNew && (
+						<Icon icon={ sparkles } style={ styles.newIndicator } />
+					) }
+					<View style={ modalIconStyle }>
+						<Icon
+							icon={ item.icon.src || item.icon }
+							fill={ modalIconStyle.fill }
+							size={ modalIconStyle.width }
+						/>
 					</View>
-					<Text numberOfLines={ 3 } style={ modalItemLabelStyle }>
-						{ blockTitle }
-					</Text>
 				</View>
-			</TouchableHighlight>
-		);
-	}
+				<Text numberOfLines={ 3 } style={ modalItemLabelStyle }>
+					{ blockTitle }
+				</Text>
+			</View>
+		</TouchableHighlight>
+	);
 }
 
-const InserterButton = withPreferredColorScheme( MenuItem );
+const InserterButton = withPreferredColorScheme( MenuItemFunction );
```

Comment:
I think there's a `PreferredColorScheme` hook that can be utilized here since we are doing this conversion. I will leave this decision up to you 😄